app/assets/javascripts/govuk_publishing_components/analytics/auto-scroll-tracker.js

```diff
@@ -203,15 +213,24 @@ window.GOVUK.Modules = window.GOVUK.Modules || {};
 
   // check heading is inside allowed elements, generally ignores everything outside of page content
   AutoScrollTracker.Heading.prototype.findAllowedHeadings = function () {
-    var headings = ['h1', 'h2', 'h3', 'h4', 'h5', 'h6']
     var headingsFound = []
+    var headings = ['h1', 'h2', 'h3', 'h4', 'h5', 'h6']
+    var trackSpecificHeadings = this.config.trackSpecificHeadings
 
+    // this is a loop that only happens once as we currently only have one
+    // allowed element for headings to be in - 'main'
```

Comment:
Yeah, me too. I put this in so that we could potentially extend it in future - for example if we needed to track headings outside of the `main` element. I might just take it out and hardwire the `main` element though, will have a think.